src/modules/twap/utils/getTwapOrderStatus.ts

```diff
@@ -11,19 +11,17 @@ export function getTwapOrderStatus(
   { confirmedPartsCount, info: executionInfo }: TwapOrdersExecution
 ): TwapOrderStatus {
   const isFulfilled = isTwapOrderFulfilled(order, executionInfo.executedSellAmount)
+  const isCancelled = auth === false
+  const isExpired = confirmedPartsCount === order.n || isTwapOrderExpired(order, executionDate)
```

Comment:
For example, we have a TWAP order with 3 parts:
1. Expired
2. Cancelled
3. Fulfilled

The deadline for TWAP order is 13:00, the last part was filled at 12:30.
To not display a TWAP order as open for the last 30 minutes, we automatically mark it as Expired.

Reply:
I also don't get it, if the order was fulfilled, it won't show as open, right, it will show as filled. Why in this case you would want to show expired?

Reply:
<img width="1124" alt="image" src="https://github.com/cowprotocol/cowswap/assets/7122625/df867e2d-4c19-40f6-adb4-0a4271723cb8">

Reply:
During this peach color timeline (12:25 - 12:30) there are no open parts and the parent is not fulfilled (only 1 part is filled). To not keep it open we mark it as expired.

Reply:
Thanks for the diagram, makes it much clearer!

As for the status.
It checks whether all of the orders have been confirmed (and that's the same as created on the backed?), but does not check their status. That's what's still not clear to me.

Or, does that mean: by now all orders that should have been created have already been created?

Reply:
>and that's the same as created on the backed?

No, confirmed means status in [Expired, Cancelled, Fulfilled]. I've inherited the naming from the existing const `CONFIRMED_STATES`

Reply:
So, the exact code:

```
export const CONFIRMED_STATES = [OrderStatus.FULFILLED, OrderStatus.EXPIRED, OrderStatus.CANCELLED, OrderStatus.FAILED]

const confirmedPartsCount = children.filter((order) => CONFIRMED_STATES.includes(order.status)).length
```

Reply:
Ah, got it!
Confirmed means (in my head) final states.
So when all parts are finalized, the `isExpired` is set to true, but it won't be a problem with fulfilled because it takes precedence, right?